api/src/auth/drivers/oauth2.ts

```diff
@@ -281,7 +282,7 @@ export function createOAuth2AuthRouter(providerName: string): Router {
 				});
 			} catch (error: any) {
 				// Prompt user for a new refresh_token if invalidated
-				if (error instanceof InvalidTokenException && !prompt) {
+				if ((error instanceof TokenExpiredException || error instanceof InvalidTokenException) && !prompt) {
```

Comment:
This shouldn't be required for Directus token handling. This code is specific to when the OAuth "refresh_token" expires, and in that instance we need to trigger a special authorization from the user to get a new one.

Reply:
You're right, I was too fixated on the exception itself rather than the context of when it actually happens 🤦‍♂️